torch/export/_trace.py

```diff
@@ -1819,7 +1827,9 @@ def _produce_guards_callback(gm):
             _is_torch_jit_trace=_is_torch_jit_trace,
         )
 
-    with fake_mode, _DataDependentErrorHandlerNonStrict():
+    with fake_mode, _DataDependentErrorHandlerNonStrict(), torch._dynamo.config.patch(
```

Comment:
Don't we only set this to False if there is some shape marked with "True"?

Reply:
Ah no, this PR redoes the whole spec, creating static constraints for everything not marked True or Dim